### PR TITLE
Settings based Comment Start at line beginning and Blank line commenting

### PR DESCRIPTION
**Pull request by oirfeodent**
Hi,
Requested for some help with Line Comment start position and seems natively there is no support.
Link of the discussion.
https://notepad-plus-plus.org/community/topic/12449/line-comment-position/

So, created a merge request for the same.
Let me know, if some thing is missing.
Regards,

### CODE REVIEW

File: PowerEditor/src/Notepad_plus.cpp [hunk 1]
```diff
@@ -3957,10 +3958,10 @@ bool Notepad_plus::doBlockComment(comment_mode currCommentMode)
 		size_t lineEnd = _pEditView->execute(SCI_GETLINEENDPOSITION, i);
 
 		// empty lines are not commented
-		if (lineIndent == lineEnd)
+		if (lineIndent == lineEnd && !nppGUI._lineCoomentIncBlankLines)
```
Comment (MAPJe71) [inline PowerEditor/src/Notepad_plus.cpp]: You misspelled `_lineCommentIncBlankLines`.
But `_lineCommentBlankLines` is more consistent with the description of the parameter and `Inc` might be associated with *increment* instead of *include*.

File: PowerEditor/src/WinControls/Preference/preference.rc [hunk 2]
```diff
@@ -112,6 +112,9 @@ BEGIN
     CONTROL         "Enable smooth font",IDC_CHECK_SMOOTHFONT,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,298,146,129,10
     CONTROL         "Enable scrolling beyond last line",IDC_CHECK_SCROLLBEYONDLASTLINE, "Button",BS_AUTOCHECKBOX | WS_TABSTOP,298,159,139,10
     CONTROL         "Disable advanced scrolling feature\r(if you have touchpad problem)",IDC_CHECK_DISABLEADVANCEDSCROLL, "Button",BS_AUTOCHECKBOX | BS_MULTILINE | WS_TABSTOP,298,172,139,18
+    GROUPBOX        "Comment Settings",IDC_COMMENT_SETTINGS_GB_STATIC,142,152,148,30,BS_CENTER
+    CONTROL         "Comment at start of Line",IDC_CHECK_LINE_COMMENT_AT_START_OF_LINE,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,153,160,120,10
+    CONTROL         "Comment Blank Lines",IDC_CHECK_LINE_COMMENT_BLANK_LINE,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,153,170,120,10
```
Comment (MAPJe71) [inline PowerEditor/src/WinControls/Preference/preference.rc]: Naming it IDC_CHECK_LINE_COMMENT_BLANK_LINE**S** would be more consistent with the description.

File: PowerEditor/src/Notepad_plus.cpp (hunk 1 above)
Comment (oirfeodent) [inline PowerEditor/src/Notepad_plus.cpp]: Made another commit 7e9f359, to make these changes.
However, I am not able to see them on the pull request 2394.
Should I delete this pull request and recreate another pull request?

Regards,

File: PowerEditor/src/WinControls/Preference/preference.rc (hunk 2 above)
Comment (oirfeodent) [inline PowerEditor/src/WinControls/Preference/preference.rc]: Changed with the second commit 7e9f359

**Comment by dail8859** [comment]
Personally I'm not a fan of a global setting that covers all languages. Some languages *require* the comment is at the start of a line (not sure if any require it *not* be on the start of a line). 

Also, from looking at the code it doesn't seem like these settings would be persistent.

**Comment by oirfeodent** [comment]
ok @dail8859
I will close this pull request and move this as a Lexer specific.
I started this as a generic one because, I am using UDL as the Lexer I want is not yet there in NPP.

I will check on adding the Lexer first and then check on this request.

@MAPJe71 & @dail8859 ,
Thx for your time & response so far.

**Comment by dail8859** [comment]
@oirfeodent I'm not saying this is the wrong way of doing it. I'm honestly not sure how hard it would be to make it a per languages setting (it might be quite difficult and not worth it). This may well be a far easier way of doing it that would probably help out a lot of users. Current there is *no* way of controlling comment position, at least this gives the user some control over it :)

**Comment by oirfeodent** [comment]
@dail8859 , I am as well not sure on per language setting. Anyway, I will try to get my Lexer into NPP first, after (and if) scintilla gets updated to 3.7.0. If this feature seems difficult with per language setting, I will reopen/create another pull request.

**Comment by Nightblade** [comment]
Any chance you might have time to take another look at this? 

Related:  https://github.com/notepad-plus-plus/notepad-plus-plus/issues/1962  https://github.com/notepad-plus-plus/notepad-plus-plus/issues/706
